packages/shared/src/contexts/AuthContext.tsx

```diff
@@ -162,24 +160,25 @@ export const AuthContextProvider = ({
         firstVisit: user?.firstVisit,
         trackingId: user?.id,
         shouldShowLogin: loginState !== null,
-        showLogin: useCallback(
-          ({ trigger, options = {} }) => {
-            const hasCompanion = !!isCompanionActivated();
-            if (hasCompanion) {
-              const signup = `${process.env.NEXT_PUBLIC_WEBAPP_URL}signup?close=true`;
-              window.open(signup);
-            } else {
-              const params = new URLSearchParams(globalThis?.location.search);
+        showLogin: useCallback(({ trigger, options = {} }) => {
+          const hasCompanion = !!isCompanionActivated();
+          if (hasCompanion) {
+            const signup = `${process.env.NEXT_PUBLIC_WEBAPP_URL}signup?close=true`;
+            window.open(signup);
+          } else {
+            const params = new URLSearchParams(globalThis?.location.search);
 
-              setLoginState({ ...options, trigger });
-              if (!params.get(AFTER_AUTH_PARAM)) {
-                params.set(AFTER_AUTH_PARAM, window.location.pathname);
-              }
-              router.push(`/onboarding?${params.toString()}`);
+            setLoginState({ ...options, trigger });
+            if (!params.get(AFTER_AUTH_PARAM)) {
+              params.set(AFTER_AUTH_PARAM, window.location.pathname);
             }
-          },
-          [router],
-        ),
+            window.history.pushState(
```

Comment:
I had to get rid of router to make it work on storybook. Apparently introducing router here broke many stories we had

Reply:
We should use router window.history does not work well with Next.js pages, we already have issues on post page where we use it on modal from feed. Especially on back button. Don't have any concrete examples but I am sure its gonna be a problem.

Reply:
Ok so I'll try to mock router on storybook side 👌

Reply:
Fixed in https://github.com/dailydotdev/apps/pull/4355/commits/0cdaf06d6aeea5376fe7d320e2be8c559e8d461b

Reply:
very nice 🚀